### DISCUSSION TITLE
File Import not supported for Reentrancy Guard

### DISCUSSION THREAD

**Discussion by therevealer** [Cyfrin/foundry-full-course-cu#2682] [Q&A] [answered]
In the Advanced foundry course on the Defi section, Patrick imported the reentrancy guard from openzeppelin using this format
`import { ReentrancyGuard } from "@openzeppelin/contracts/security/ReentrancyGuard.sol";`

Following this my Vscode cannot locate the file, I have also confirmed my `foundry.toml` for the remappings.
```
[profile.default]
src = "src"
out = "out"
libs = ["lib"]
remappings = ["@openzeppelin/contracts=lib/openzeppelin-contracts/contracts"]
```
I tried checking the rentrancy file manually and I found it in a Javascript file

I wrote the import manually but still getting an error
`import {ReentrancyGuard} from "@openzeppelin/contracts/test/utils/ReentrancyGuard.sol";`
Can foundry make use of js files?

<img width="1067" alt="Screen Shot 2024-10-15 at 1 44 28 AM" src="https://github.com/user-attachments/assets/f914009b-c9fc-42ad-8017-d24567fc3fe6">

**Comment by EngrPips** [comment]
Hello @therevealer, When you run `forge build` does you code compile successfully?

**Comment by therevealer** [comment]
@EngrPips Here is the error i get when i run`forge build`

Using this `import { ReentrancyGuard } from "@openzeppelin/contracts/security/ReentrancyGuard.sol";`

```
Setups-MacBook-Pro:Defi - Stablecoins Revealer$ forge build
[⠃] Compiling...2024-10-15T17:20:48.745505Z ERROR foundry_compilers_artifacts_solc::sources: error="/Users/setup/Desktop/security/security skillup/Defi - Stablecoins/lib/openzeppelin-contracts/contracts/security/ReentrancyGuard.sol": No such file or directory (os error 2)
[⠊] Compiling...
Error: 
failed to resolve file: "/Users/setup/Desktop/security/security skillup/Defi - Stablecoins/lib/openzeppelin-contracts/contracts/security/ReentrancyGuard.sol": No such file or directory (os error 2); check configured remappings
        --> /Users/setup/Desktop/security/security skillup/Defi - Stablecoins/src/DSCEngine.sol
        @openzeppelin/contracts/security/ReentrancyGuard.sol
```

**Reply by EngrPips** [reply, marked as answer]
Please press `command + p` then type `ReentrancyGuard` in the search bar that appears and `vscode` will direct you to the location of the `ReentrancyGuard` file and you can use that path to import the file.

**Reply by therevealer** [reply]
Done ser

Code compiles successfully 🥳

**Reply by EngrPips** [reply]
Awesome, Keep crushing it!
